BridgeCareApp/BridgeCareCodeFirst/Models/BudgetReportData.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Data.SqlClient;
+using System.Linq;
+using static BridgeCare.Models.BudgetReportData;
+
+namespace BridgeCare.Models
+{
+    public class BudgetReportData : IBudgetReportData
+    {
+        private readonly List<ICostDetails> costList;
+
+        public BudgetReportData()
+        {
+        }
+        public BudgetReportData(List<ICostDetails> costList)
+        {
+            this.costList = costList ?? throw new ArgumentNullException(nameof(costList));
+        }
+        public BudgetReportDetails GetBudgetReportData(SimulationResult data, string[] budgetTypes)
+        {
+            var BudgetYearView = new Hashtable();
+            var db = new BridgeCareContext();
+            string getReport =
+                "SELECT Years, Budget, Cost_ " +
+                    " FROM Report_" + data.NetworkId
+                    + "_" + data.SimulationId + " WHERE BUDGET is not null";
+            try
+            {
+                var RawQueryForData = db.Database.SqlQuery<BudgetReport>(getReport).AsQueryable();
+
+                double sum = 0;
+                foreach (var report in RawQueryForData)
+                {
+                    double.TryParse(report.Cost_.Value.ToString(), out double cost);
+                    Hashtable yearView;
+                    if (!BudgetYearView.Contains(report.Budget))
+                    {
+                        yearView = new Hashtable();
+                        BudgetYearView.Add(report.Budget, yearView);
+                    }
+                    else
+                    {
+                        yearView = (Hashtable)BudgetYearView[report.Budget];
+                    }
+
+                    if (yearView.Contains(report.Years))
+                    {
+                        sum = (double)yearView[report.Years];
+                        yearView.Remove(report.Years);
+                    }
+                    sum += cost;
+                    yearView.Add(report.Years, sum);
+                    costList.Add(new CostDetails
+                    {
+                        Cost = report.Cost_.Value,
+                        Years = report.Years,
+                        Budget = report.Budget
+                    });
+                }
+                foreach (string item in BudgetYearView.Keys)
```

Comment:
Add a comment for this loop describing what is happening.

Reply:
added